api4/channel_category_test.go

```diff
@@ -90,6 +91,17 @@ func TestCreateCategoryForTeamForUser(t *testing.T) {
 		require.Equal(t, int64(10), customCategory.SortOrder)
 	})
 
+	t.Run("should not crash with null input", func(t *testing.T) {
+		require.NotPanics(t, func() {
+			user, client := setupUserForSubtest(t, th)
+			payload := []byte(`null`)
+			route := fmt.Sprintf("/users/%s/teams/%s/channels/categories", user.Id, th.BasicTeam.Id)
+			r, err := client.DoAPIPostBytes(route, payload)
+			require.Error(t, err)
+			defer closeBody(r)
```

Comment:
Why is there a `defer` on the last line of the function?

Reply:
heh good point. Copy pasta :)